### PR TITLE
Updates worker to work with Instagram signature contents changes and …

### PR DESCRIPTION
**Pull request by krhoda**
…FB deletion policy requirements

### CODE REVIEW

File: worker/worker/worker.js [hunk 1]
```diff
@@ -247,11 +250,20 @@ async function handler_instagram_login(request) {
   }
 }
 
+async function handler_data_deletion(request) {
+  return new Response(JSON.stringify({
+    code: `${+ new Date.now()}`,
```
Comment (sbihel) [inline worker/worker/worker.js:255]: What does this do?

File: worker/worker/worker.js [hunk 2]
```diff
@@ -247,11 +250,20 @@ async function handler_instagram_login(request) {
...
+  }))
+}
+
 async function handleRequest(request) {
   const r = new Router();
   r.get('/witness_tweet', request => handler_witness_tweet(request));
   r.get('/witness_instagram_post', request => handler_witness_instagram_post(request));
   r.get('/instagram_login', request => handler_instagram_login(request));
+  r.get('/instagram_data_deletion', request => handler_data_deletion(request))
+  r.get('/instagram-deauth', request => handler_data_deletion(request))
```
Comment (sbihel) [inline worker/worker/worker.js:266]: Are these endpoints defined by FB?

File: worker/worker/worker.js [hunk 3]
```diff
@@ -201,6 +201,9 @@ async function handler_witness_instagram_post(request) {
     }
 
     let kvObj = JSON.parse(kvEntry);
+    if (!kvObj?.link || !kvObj?.sig) {
```
Comment (sbihel) [inline worker/worker/worker.js:204]: Can that happen?

**Review by sbihel — APPROVED** [review]

File: worker/worker/worker.js (hunk 1 above)
Comment (krhoda) [reply worker/worker/worker.js:255]: It it supposed to provide a confirmation that the authorization data is deleted, but we don't store any. This conforms to the FB requirements, as far as I can tell. Most importantly, I direct the end user to a page that states we don't store any authorization data.

File: worker/worker/worker.js (hunk 2 above)
Comment (krhoda) [reply worker/worker/worker.js:266]: Yes, in the sense I need to provide something for "data deletion" and "deauth". It's stated in the docs that if you don't store anything you don't have to do anything, but they have to exist.

File: worker/worker/worker.js (hunk 3 above)
Comment (krhoda) [reply worker/worker/worker.js:204]: If given a non-existent handle,`{}` comes from the KV store.

**Pull request merged**